### PR TITLE
-Use the Context2D directly to get the pixel color, instead of making…

### PR DESCRIPTION
**Pull request by mdproctor**
… a copy of the ImageData (which turns out to be expensive and uneeded).

### CODE REVIEW

**Comment by romartin** [comment]
@tiagodolphine FYI

File: src/main/java/com/ait/lienzo/client/core/shape/wires/picker/ColorMapBackedPicker.java [hunk 1]
```diff
@@ -118,10 +116,10 @@ public PickerPart findShapeAt(int x, int y)
             y = (int) Math.round(temp.getY());
         }
 
-        String color = BackingColorMapUtils.findColorAtPoint(m_imageData, x, y);
+        ImageDataPixelColor color = m_ctx.getImageDataPixelColor(x, y);
```
Comment (hasys) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/picker/ColorMapBackedPicker.java]: It looks like after `color = Context2D#getImageDataPixelColor(x, y);` always follow `color.toBrowserRGB` with one small exception - in one place `return null if color.A is not 255 else color.toBrowserRGB`;

Do we need to have `ImageDataPixelColor` class and create it's objects?

File: src/main/java/com/ait/lienzo/client/core/shape/wires/picker/ColorMapBackedPicker.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import com.ait.lienzo.client.core.shape.wires.WiresShape;
 import com.ait.lienzo.client.core.types.ColorKeyRotor;
 import com.ait.lienzo.client.core.types.ImageData;
```
Comment (hasys) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/picker/ColorMapBackedPicker.java]: This import statement is not in use any more.

File: src/main/java/com/ait/lienzo/client/core/shape/wires/picker/ColorMapBackedPicker.java [hunk 3]
```diff
@@ -118,10 +116,10 @@ public PickerPart findShapeAt(int x, int y)
             y = (int) Math.round(temp.getY());
         }
 
-        String color = BackingColorMapUtils.findColorAtPoint(m_imageData, x, y);
+        ImageDataPixelColor color = m_ctx.getImageDataPixelColor(x, y);
         if (color != null)
         {
-            PickerPart pickerPart = m_colorMap.get(color);
+            PickerPart pickerPart = m_colorMap.get(color.toBrowserRGB());
             if (pickerPart != null)
```
Comment (hasys) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/picker/ColorMapBackedPicker.java]: This `If` statement is redundant, method will return `null` in both cases: if `pickerPart is null` and if `pickerPart is not null` as well.

**Comment by romartin** [comment]
Hey @tiagodolphine 

You think can we close this PR and just reference the new one that includes all commits as well? 
https://github.com/kiegroup/lienzo-core/pull/52/

If so, please take into account @hasys review and comments.

Thanks!

**Comment by hasys** [comment]
Hi @romartin,

in general it is easier to deal with smaller PRs, but if it is already merged to one mega PR I guess we have no other choose than close this PR.

**Comment by romartin** [comment]
@tiagodolphine ok on closing this?

**Comment by tiagodolphine** [comment]
@romartin yes this could be closed since the changes referred by this PR will be merged on https://github.com/kiegroup/lienzo-core/pull/52. You can check the commits there.

**Pull request closed without merging**